pkg/batchd/cache/cache.go

```diff
@@ -88,6 +89,9 @@ func newSchedulerCache(config *rest.Config) *SchedulerCache {
 			FilterFunc: func(obj interface{}) bool {
 				switch t := obj.(type) {
 				case *v1.Pod:
+					if strings.Compare(obj.(*v1.Pod).Spec.SchedulerName, schedulerName) != 0 {
```

Comment:
Only filter Pods with the same scheduler name may be not enough. It may cause `kube-batchd` schedule pods to a node overload.

Such as some pods from another scheduler are running on a node, however, the node is idle for `kube-batchd` due to `kube-batchd` doesn't get those running pods info. When `kube-batchd` try to schedule pods on that node, it may fail due to lack of resources.

Reply:
+1

We should watch 'pending & schedulerName' or 'running' pods.

Reply:
@jinzhejz @k82cn thanks for your review. I'll update the condition. How about 'pending & schedulerName' or 'nonTerminatedPod()'?